### PR TITLE
refs task #39197 Register

### PR DESCRIPTION
**Pull request by nvvu99**
## Related Tickets
- [#39197](https://edu-redmine.sun-asterisk.vn/issues/39197)

## WHAT (optional)
Add registration feature for employee and employer

## HOW

Create RegisterController with the following methods:
* showRegistrationForm to return a view with registration forms.
* registerEmployee to handle request to register an employee user.
* registerEmployer to handle request to register an employer user.

Assign routes for these above actions.

Validate the request input using form request.
When received a request with valid input, create a user and a profile for this user.
Set the status of employee user to 'activated'.
Set the status of employer user to 'inactivated'.

Update localization for login and register pages.

## Evidence (Screenshot or Video)
* Employee registration
Register

* Employer registration
Employer register

### CODE REVIEW

File: resources/lang/vi/messages.php [hunk 1]
```diff
@@ -30,4 +30,10 @@
     'make-cv' => 'Tạo CV',
     'best-offer' => 'Cung cấp các dịch vụ hàng đầu',
     'industry' => 'Lĩnh Vực',
+    'remember' => 'Ghi nhớ',
+    'have-no-account' => 'Chưa có tài khoản?',
+    'register-now' => 'Đăng ký ngay!',
+    'employee-registered' => 'Đăng ký tài khoản thành công!',
+    'employer-registered' =>
+        'Đăng ký tài khoản thành công! Vui lòng chở xác nhận của admin.',
```
Comment (vuminhhieu1311) [inline resources/lang/vi/messages.php:38]: *chờ xác nhận

File: resources/lang/vi/validation.php [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    'string' => ':Attribute phải là chuỗi ký tự.',
+    'digits' => ':Attribute phải có :digits chữ số.',
+    'starts_with' => ':Attribute phải bắt đầu với: :values.',
+    'url' => ':Attribute không phải URL hợp lệ.',
+
+    'attributes' => [
+        'name' => 'tên',
+        'password' => 'mật khẩu',
+        'address' => 'địa chỉ',
+        'industry' => 'lĩnh vưc',
```
Comment (vuminhhieu1311) [inline resources/lang/vi/validation.php:23]: *lĩnh vực :v

**Review by vuminhhieu1311 — APPROVED** [review]

**Review by vuminhhieu1311 — APPROVED** [review]

**Comment by nvvu99** [comment]
ready

File: app/Http/Requests/Auth/EmployeeRegisterRequest.php [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace App\Http\Requests\auth;
```
Comment (huyenttm-0901) [inline app/Http/Requests/Auth/EmployeeRegisterRequest.php:3]:
```suggestion
namespace App\Http\Requests\Auth;
```

File: app/Http/Requests/Auth/EmployerRegisterRequest.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Requests\auth;
```
Comment (huyenttm-0901) [inline app/Http/Requests/Auth/EmployerRegisterRequest.php:3]: tương tự

**Comment by nvvu99** [comment]
ready

**Comment by nvvu99** [comment]
ready

**Review by huyenttm-0901 — APPROVED** [review]

**Pull request merged**